contracts/adapters/Managing.sol

```diff
@@ -136,12 +126,42 @@ contract ManagingContract is
         );
 
         dao.processProposal(proposalId);
-        dao.replaceAdapter(
-            proposal.adapterId,
-            proposal.adapterAddress,
-            proposal.flags,
-            proposal.keys,
-            proposal.values
-        );
+        if (proposal.updateType == UpdateType.ADAPTER) {
+            dao.replaceAdapter(
+                proposal.adapterOrExtensionId,
+                proposal.adapterOrExtensionAddr,
+                proposal.flags,
+                proposal.keys,
+                proposal.values
+            );
+        } else if (proposal.updateType == UpdateType.EXTENSION) {
+            if (dao.extensions(proposal.adapterOrExtensionId) != address(0x0)) {
+                dao.removeExtension(proposal.adapterOrExtensionId);
+            }
+
+            if (proposal.adapterOrExtensionAddr != address(0x0)) {
+                dao.addExtension(
+                    proposal.adapterOrExtensionId,
+                    IExtension(proposal.adapterOrExtensionAddr),
+                    msg.sender
```

Comment:
@adridadou we were using the `dao` address here, but then when we try to initialize the new extension it fails, because that address is not a member - which is correct. So I thought we should use the sender as the creator address, WDYT?

Reply:
we can just take the first member of the DAO and pass it. WDYT ?